src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets

```diff
@@ -21,7 +21,8 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <ContainerName>$(TestProduct)-$(Branch)-$(BuildMoniker)</ContainerName>
+    <TargetOS>$(TargetQueue.Replace('.', '-'))</TargetOS>
+    <ContainerName>$(TestProduct)-$(Branch)-$(BuildMoniker)-$(TargetOS)</ContainerName>
```

Comment:
Could you please add a condition to ContainerName that allows it to be overridden (see RunnerScript for an example)?